### DISCUSSION TITLE
Code organisation help

### DISCUSSION THREAD

**Discussion by Wulfheart** [filamentphp/filament#11262] [Help]
### Package

Panel builder

### Package Version

v3.2.0

### How can we help you?

Hello,

I am currently unsure how I should organize my code to work better. I want to display different edit/view pages for different states a model is in. E.g. my `Game` model has the states `PreGame`, `Playing`, `Finished` with different RelationManagers each. Is there a way to have multiple edit pages with different RelationManagers each? I was unfortunately unable to find docs on this.

Thank you in advance.

**Comment by micraux** [comment]
Hello,

I don't understand your logic here: what do you mean by "_my Game model has the states PreGame, Playing, Finished with different RelationManagers each_"?

I assume "_states_" is a `state` (enum) column in your `Game` model...

`RelationManagers` are made for linking Models thanks to relationships, nothing to do with _states_ here.

You can have multiple edit pages for one resource and use a query in each edit page to retrieve your records based on their state for example... Is that what you want to achieve?
